### PR TITLE
chore!: Delete unused code in feature view

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

BREAKING CHANGE: Deprecate feature view parameters such as `batch_source` and `stream_source` and deprecate the option to use positional arguments

### CODE REVIEW

File: sdk/python/feast/on_demand_feature_view.py [hunk 1]
```diff
@@ -94,136 +92,29 @@ def __init__(  # noqa: C901
...
-                _udf = args[3]
-
-        if not _name:
-            raise ValueError(
-                "The name of the on demand feature view must be specified."
-            )
-
-        if not _sources:
-            raise ValueError("The `sources` parameter must be specified.")
-
```
Comment (achals) [inline sdk/python/feast/on_demand_feature_view.py:215]: 💯

**Review by achals — COMMENTED** [review]
Is this still a WIP PR? Looks pretty good to me.

File: sdk/python/feast/feature_view.py [hunk 2]
```diff
@@ -61,10 +60,10 @@ class FeatureView(BaseFeatureView):
         ttl: The amount of time this group of features lives. A ttl of 0 indicates that
             this group of features lives forever. Note that large ttl's or a ttl of 0
             can result in extremely computationally intensive queries.
-        batch_source (optional): The batch source of data where this group of features
+        batch_source: The batch source of data where this group of features
```
Comment (adchia) [inline sdk/python/feast/feature_view.py:63]: huh did you do this in reverse? the docs suggest we want source instead of batch_source vs stream_source

Comment (adchia) [reply sdk/python/feast/feature_view.py:63]: oh these are attributes. Maybe remove the two lines on "deprecated in favor of source"

Comment (felixwang9817) [reply sdk/python/feast/feature_view.py:63]: @adchia the `__init__` takes in `source` as a parameter; it then sets `batch_source` and `stream_source` as attributes

**Review by adchia — APPROVED** [review]
just a nit

File: sdk/python/feast/feature_view.py (hunk 2 above)
Comment (felixwang9817) [reply sdk/python/feast/feature_view.py:63]: good catch

Comment (adchia) [reply sdk/python/feast/feature_view.py:63]: yep. just remove the two lines here saying "This is deprecated in favor of `source`" ?

**Review by adchia — APPROVED** [review]
/lgtm

**Pull request merged**